### PR TITLE
Add connector to Galoy wallets

### PR DESCRIPTION
**Pull request by bumi**
PR by @vindard 
This allows Alby to connect to [Galoy](https://galoy.io/) wallets.
(original branch: https://github.com/vindard/lightning-browser-extension/tree/feature/galoy-connector)

### CODE REVIEW

File: src/app/screens/Onboard/ConnectGaloy/index.tsx [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+const defaultHeaders = {
+  Accept: "application/json",
+  "Access-Control-Allow-Origin": "*",
+  "Content-Type": "application/json",
+};
+
+export default function ConnectGaloy() {
+  const navigate = useNavigate();
+  const [loading, setLoading] = useState(false);
+  const [phoneNumber, setPhoneNumber] = useState<string>("");
```
Comment (dylancom) [inline src/app/screens/Onboard/ConnectGaloy/index.tsx:19]: Typescript will infer the type as string while passing "", so no need to add "<string>"

File: src/app/screens/Onboard/ConnectGaloy/index.tsx [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+  const [smsCode, setSmsCode] = useState<string | undefined>();
+  const [smsCodeRequested, setSmsCodeRequested] = useState<
```
Comment (dylancom) [inline src/app/screens/Onboard/ConnectGaloy/index.tsx:21]: We can just rely on boolean and set it to false initially.

File: src/app/screens/Onboard/ConnectGaloy/index.tsx [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+  Accept: "application/json",
+  "Access-Control-Allow-Origin": "*",
+  "Content-Type": "application/json",
+};
+
+export default function ConnectGaloy() {
+  const navigate = useNavigate();
+  const [loading, setLoading] = useState(false);
+  const [phoneNumber, setPhoneNumber] = useState<string>("");
+  const [smsCode, setSmsCode] = useState<string | undefined>();
```
[the remark this answers is not in the document]
Comment (bumi) [reply src/app/screens/Onboard/ConnectGaloy/index.tsx:20]: that would be similar to phoneNumber then, right? so we should do that.

File: src/app/screens/Onboard/ConnectGaloy/index.tsx (hunk 2 above)
Comment (bumi) [reply src/app/screens/Onboard/ConnectGaloy/index.tsx:21]: +1 only boolean. it is either requested or not.

File: src/app/screens/Onboard/ChooseConnector/index.tsx [hunk 4]
```diff
@@ -32,6 +33,11 @@ export default function ChooseConnector() {
       description: "Connect to your LNbits account",
       logo: lnbits,
     },
+    {
+      to: "galoy",
+      title: "Galoy",
+      description: "Create or connect to a Galoy account",
```
Comment (bumi) [inline src/app/screens/Onboard/ChooseConnector/index.tsx:39]: we need a logo here now.

**Comment by vindard** [comment]
Just a quick comment, there's still very small changes happening to the API that can potentially break things. Once this stabilizes (expected in the next week or 2), I'll give this another pass and adjust anything that needs fixing.

**Comment by bumi** [comment]
thanks for the update @vindard

File: src/app/screens/Onboard/ChooseConnector/index.tsx (hunk 4 above)
Comment (vindard) [reply src/app/screens/Onboard/ChooseConnector/index.tsx:39]: Added here: https://github.com/getAlby/lightning-browser-extension/pull/483/commits/00194782c331c397267311a3a774ba78185a1a9e

**Comment by vindard** [comment]
> Just a quick comment, there's still very small changes happening to the API that can potentially break things. Once this stabilizes (expected in the next week or 2), I'll give this another pass and adjust anything that needs fixing.

I just opened #483, our mobile app is now entirely switched across to the new API and rolled out to 100% of our users so this should be safe to use now. I added one small change to how we fetch the wallet-id and I refactored `balance` fetching to be a bit more resilient to some of our future planned changes.

_**Note:** The changes in this PR will now also allow users of the Bitcoin Jungle wallet (https://bitcoinjungle.app/) to use Alby._

**Comment by vindard** [comment]
Tagging another Galoy contributor @dolcalmi to potentially take a look at this if he gets the chance. Looks ok now from my end.

**Comment by vindard** [comment]
I just did a mainnet test here, all looks good so far 👌🏽
 https://twitter.com/vindaRd/status/1476778088218386433

**Comment by bumi** [comment]
@vindard super cool! nice. 

do you have time to fix the conflicts? I guess it's nothing big. 
then we can push this in the next release.

**Comment by vindard** [comment]
> do you have time to fix the conflicts? I guess it's nothing big.

Ya I started taking a look but it looks like it's some things with changed file structure.

I can take a look again over the next few days, but I suspect someone who knows the changed structure would be able to move a lot quicker if one of you guys gets to it first.

**Comment by bumi** [comment]
I merged the master. We have made some refactorings for the connectors. 

and we now need one more method: 
https://github.com/getAlby/lightning-browser-extension/pull/434/files#diff-276efcad9996c7ee83133055ffe4b0e26c8276bab84eb05b36187aca04016ac7R134-R141

that allows alby to check if a payment is received.

**Comment by vindard** [comment]
> I merged the master. We have made some refactorings for the connectors.
I started doing this as well and saw this, changes look really clean 👍🏾 

> that allows alby to check if a payment is received. I guess Galoy has such an endpoint. can you add that?
Yup sure, I can do this

File: src/app/screens/Onboard/ConnectGaloy/index.tsx [hunk 5]
```diff
@@ -0,0 +1,258 @@
+import { useState } from "react";
+import { useNavigate } from "react-router-dom";
+import axios from "axios";
+import Input from "../../../components/Form/Input";
+import Button from "../../../components/Button";
+
+import utils from "../../../../common/lib/utils";
+
+const url = process.env.GALOY_URL || "https://api.staging.galoy.io/graphql/";
```
Comment (bumi) [inline src/app/screens/Onboard/ConnectGaloy/index.tsx:9]: we should default here to the production URL

Comment (vindard) [reply src/app/screens/Onboard/ConnectGaloy/index.tsx:9]: Done: https://github.com/getAlby/lightning-browser-extension/pull/505/commits/e73f383a5264571709fc1dc4972b337031e295bf

**Comment by vindard** [comment]
Lmk if you guys need anything else on this one

**Comment by bumi** [comment]
@vindard thanks! we should get this in in one of the next releases. 
maybe @dylancom can have another look at the code. 
And we do some more tests, @MoritzKa can you also check this?

**Comment by vindard** [comment]
Incorporated suggestion from https://github.com/getAlby/lightning-browser-extension/pull/558#issuecomment-1021307631 to separate out instances in PR #637 targeted at this PR

**Comment by vindard** [comment]
Hey hey, let me know if there's anything else you guys need help with to get this across the finish line (open questions, testing etc.) 🎉

**Comment by bumi** [comment]
@johnsBeharry ack?

**Comment by johnsBeharry** [comment]
tested this out. looking sexy! also I agree with having separate connectors for each backend. people are not going to be so familiar with the Galoy brand, but the wallets that get created.

File: src/extension/background-script/connectors/connector.interface.ts [hunk 6]
```diff
@@ -14,22 +14,28 @@ export interface MakeInvoiceArgs {
   memo: string;
 }
 
-export interface MakeInvoiceResponse {
-  data: {
-    paymentRequest: string;
-    rHash: string;
-  };
-}
+export type MakeInvoiceResponse =
```
Comment (dylancom) [inline src/extension/background-script/connectors/connector.interface.ts:17]: I think our response types have to describe a happy path response. And when an error happens somewhere we should throw / catch it.

File: src/app/screens/Onboard/ConnectGaloy/index.tsx (hunk 1 above)
Comment (vindard) [reply src/app/screens/Onboard/ConnectGaloy/index.tsx:40]: Addressed in #714

File: src/app/screens/Onboard/ConnectGaloy/index.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (vindard) [reply src/app/screens/Onboard/ConnectGaloy/index.tsx:41]: Addressed in #714

File: src/app/screens/Onboard/ConnectGaloy/index.tsx (hunk 2 above)
Comment (vindard) [reply src/app/screens/Onboard/ConnectGaloy/index.tsx:42]: Addressed in #714

File: src/extension/background-script/connectors/connector.interface.ts (hunk 6 above)
Comment (vindard) [reply src/extension/background-script/connectors/connector.interface.ts:17]: Addressed in #714

**Comment by vindard** [comment]
Hey folks, I've added changes for all open comments and pulled in `master` to resolve any open merge conflicts in PR #714 targeted at this branch @bumi @dylancom 

If there's anything open on this one, please let me know so we can get this one buttoned up. Would really love to get this merged soon if possible (it's getting tricky to keep maintaining as an open fork 😅)

**Pull request merged**